src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +69,82 @@ public void TestSignaturesV2_4_X()
         {
             Assert.IsType<SparkSession>(SparkSession.Active());
         }
+
+        /// <summary>
+        /// Test CreateDataFrame API.
+        /// </summary>        
+        [Fact]
+        public void TestCreateDataFrame()
+        {            
+            // Calling CreateDataFrame with schema
+            {
+                var data = new List<GenericRow>();
+                data.Add(new GenericRow(new object[] { "Alice", 20 }));
+                data.Add(new GenericRow(new object[] { "Bob", 30 }));
+
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("Name", new StringType()),
+                    new StructField("Age", new IntegerType())
+                });                
+                
+                DataFrame df = _spark.CreateDataFrame(data, schema);
+                Assert.IsType<DataFrame>(df);
+                Assert.Equal(schema, df.Schema());
+                Row[] rows = df.Collect().ToArray();                
+                for (int i = 0; i < rows.Length; ++i)
+                {
+                    // Checking the values from the dataFrame are same as the values in given data
+                    Assert.Equal(data[i].Values, rows[i].Values);
+                }
+                Assert.Equal(data.Count, rows.Length);
+            }
+
+            // Calling CreateDataFrame<string> without schema
```

Comment:
` CreateDataFrame<string>` is not the correct syntax.

Reply:
How about CreateDataFrame(IEnumerable<string> _) ?